internal/cmd/test.go

```diff
@@ -26,17 +25,27 @@ const (
 	LOCAL_HOST      = "localhost"
 )
 
-type CommonCMDTestCases struct {
-	ExpectedExitCode int
-	ExpectedStdout   string
-	ExpectedStderr   string
+type CMDTestParameters struct {
+	ExpectedExitCode        int
+	ExpectedStdout          string
+	ExpectedStderrSubstring string
+	LogLevel                log.LogLevel
 }
 
 // Common setup for all CMD tests that require a server.
 func CMDServerTestingMain(suite *testing.M) {
+	server.StopServer()
+
+	port, err := getUnusedPort()
+	if err != nil {
+		log.Error("Failed to get an unused port.", err)
+		os.Exit(1)
```

Comment:
We don't use os.Exit() anymore, we use exit.Exit():
https://github.com/edulinq/autograder-server/blob/main/internal/exit/exit.go#L19